pkg/keystore/pkcs12/pkcs12.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkcs12
+
+import (
+	"crypto"
+	"crypto/rand"
+	"fmt"
+
+	csiapi "github.com/cert-manager/csi-driver/pkg/apis/v1alpha1"
+
+	"github.com/cert-manager/cert-manager/pkg/util/pki"
+	"github.com/cert-manager/csi-lib/metadata"
+	"software.sslmate.com/src/go-pkcs12"
+)
+
+// Create combines the inputs to a single PKCS12 keystore file.
+// Private key must be PKCS1 or PKCS8 encoded. Certificates must be PEM
+// encoded.
+func Create(meta metadata.Metadata, pk crypto.PrivateKey, chainPEM []byte) ([]byte, error) {
```

Comment:
**issue (blocking):** bloaty argument causing unnecessary dependencies

this package should not need to know anything about `github.com/cert-manager/csi-lib/metadata` or `github.com/cert-manager/csi-driver/pkg/apis/v1alpha1`. using these dependencies mean we'd have to update this code if API of the packages would change

**suggestion:**
```suggestion
func Create(password string, pk crypto.PrivateKey, chainPEM []byte) ([]byte, error) {
```

Reply:
I actually completely disagree- I don't think the caller should need to know anything about the API of PCS12.

I am actually going to change this package so that `writer` doesn't need to understand `KeyStorePKCS12EnableKey` or `KeyStorePKCS12FileKey`.

Reply:
Do you mean to always use `metadata.Metadata` in cases like these in `csi-driver`?

Do you mean that future programmers that want to use the public API of `csi-driver` should call https://github.com/cert-manager/csi-lib/blob/dd8ae5d66f5301940c865704f3e91992ab3bc4d6/storage/interfaces.go#L50 and then use that as the argument here, and with similar functions?